test/feedhandler.cpp

```diff
@@ -0,0 +1,155 @@
+#include "3rd-party/catch.hpp"
+
+#include <memory>
+
+#include "cache.h"
+#include "configcontainer.h"
+#include "feedhandler.h"
+#include "rss.h"
+#include "test-helpers.h"
+
+using namespace newsboat;
+
+namespace {
+
+std::vector<std::shared_ptr<rss_feed>> get_five_empty_feeds()
+{
+	std::vector<std::shared_ptr<rss_feed>> feeds;
+	TestHelpers::TempFile dbfile;
+	std::unique_ptr<configcontainer> cfg(new configcontainer());
+	std::unique_ptr<cache> rsscache(new cache(dbfile.getPath(), cfg.get()));
+	for (int i = 0; i < 5; ++i) {
+		const auto feed = std::make_shared<rss_feed>(rsscache.get());
+		feeds.push_back(feed);
+	}
+	return feeds;
+}
+
+} // anonymous namespace
+
+TEST_CASE("Correctly returns desired feed")
```

Comment:
Also, please tag the tests with class name, like this:

```c++
TEST_CASE("test description", "[feedhandler]")
```

It lets us run some specific parts of the test suite.

Reply:
Yes, forgot about this, thanks.